packages/www/server.js

```diff
@@ -1,47 +1,24 @@
-// Programmatic entrypoint into the Next.js server for `pkg` purposes. Kind of
-// hacky. Really this should just get compiled to static JS and included that
-// way. But this was expedient for livepeer-in-a-box.
-//
-// On the off chance you're here to add more command line parameters in this
-// file, you should figure out how to reuse the logic in
-// ../api/src/yargs-to-mist.ts.
+// Programmatic entrypoint into the Next.js server for `pkg` purposes.
 
-const yargs = require("yargs");
-const nextConfig = require("./next.config");
-const { createServer } = require("http");
 const next = require("next");
-const pkg = require("./package.json");
+const config = require("./next.config.js");
+const { normalizeConfig } = require("next/dist/server/config-shared");
+const {
+  PHASE_PRODUCTION_SERVER,
+} = require("next/dist/shared/lib/constants.js");
 
-const app = next({ dir: __dirname, conf: nextConfig, dev: false });
-const handle = app.getRequestHandler();
-const cli = yargs.options({
-  port: {
-    type: "number",
-    default: 3012,
-  },
-}).argv;
-
-if (cli.j) {
-  const description = {
-    name: "livepeer-www",
-    friendly: "Livepeer Dashboard",
-    desc: "Livepeer Next.js Dashboard Front-end Server",
-    optional: {
-      port: {
-        option: "--port",
-        default: 3012,
-        help: "Port to run on",
-      },
-    },
-    version: pkg.version,
-  };
-  console.log(JSON.stringify(description));
-  process.exit(0);
-}
-
-app.prepare().then(() => {
-  createServer(handle).listen(cli.port, (err) => {
-    if (err) throw err;
-    console.log(`> Ready on http://localhost:${cli.port}`);
+const getApp = async () => {
```

Comment:
I have no idea what's going on around here, first time I see this file lol

Reply:
It's this! Except now we're exporting a handler function instead of an actual HTTP server. https://nextjs.org/docs/pages/building-your-application/configuring/custom-server